src/platforms/hosted/bmp_libusb.c

```diff
@@ -47,291 +106,373 @@ void libusb_exit_function(bmp_info_s *info)
 	}
 }
 
-static bmp_type_t find_cmsis_dap_interface(libusb_device *dev, bmp_info_s *info)
+char *get_device_descriptor_string(libusb_device_handle *handle, uint16_t string_index)
 {
-	bmp_type_t type = BMP_TYPE_NONE;
+	char read_string[128] = {0};
+	const int length =
+		libusb_get_string_descriptor_ascii(handle, string_index, (unsigned char *)read_string, sizeof(read_string));
```

Comment:
`(unsigned char *)read_string` -> `(uint8_t *)read_string` would be fine here in a practical sense

Reply:
Edit as requested